src/models/CampaignPledge.ts

```diff
@@ -0,0 +1,84 @@
+import type { Types, Model, PopulatedDoc, Document } from "mongoose";
+import { Schema, model, models } from "mongoose";
+import type { InterfaceFundCampaign } from "./FundCampaign";
+import type { InterfaceUser } from "./User";
+
+/**
+ * This is an interface representing a document for a Campaign Pledge in the database(MongoDB).
+ */
+
+export interface InterfaceCampaignPledge {
+  _id: Types.ObjectId | string;
+  creatorId: PopulatedDoc<InterfaceUser & Document>;
+  members: PopulatedDoc<InterfaceUser & Document>[];
+  campaigns: PopulatedDoc<InterfaceFundCampaign & Document>[];
```

Comment:
How many members and campaigns is a CampaignPledge supposed to hold? Will it be a small number like 10-15, or can it be something infinitely scalable (that is something in the order of let's say 5% of members of the organization)? If it's the later, we should be using a joining table and not the direct embedding of ducments.